WcaOnRails/app/controllers/persons_controller.rb

```diff
@@ -6,16 +6,14 @@ def index
     respond_to do |format|
       format.html
       format.js do
-        persons = Person.joins("JOIN Countries ON countryId = Countries.id")
+        persons = Person.order(:name, :countryId)
         if params[:region] != "all"
-          persons = persons.where("countryId = :region OR continentId = :region", region: params[:region])
```

Comment:
Sounds good to me. We'll leave it for @Baiqiang to decide what the best, universal solution is.